htdocs/class/xoopskernel.php

```
@@ -192,8 +192,145 @@ public function themeSelect()
         if ($themeSelect !== '' && in_array($themeSelect, xoops_getConfigOption('theme_set_allowed'))) {
             xoops_setConfigOption('theme_set', $themeSelect);
             $_SESSION['xoopsUserTheme'] = $themeSelect;
+
+            $redirectUrl = $this->getThemeRedirectUrl();
+            if ($redirectUrl !== '') {
+                header('Location: ' . $redirectUrl, true, 303);
+                exit;
```

PHPMD (2.15.0)

[error] The method themeSelect() contains an exit expression. (undefined)

(ExitExpression)